merlin/core/dataset.py

```diff
@@ -1158,4 +1158,126 @@ def _import_positions(self, positionFileName):
+
+        Args:
+            MetaDataSetName: A name to use for this metadataset
+            MERFISHDataSets: the relative path to the json file containing
+                    the merfish datasets to use. The json file can be provided
+                    in the metadatasets subdirectory of analysis parameters, or
+                    a full path can be provided to another directory.
+        """
+
+        super().__init__(metaDataSetName)
+
+        self.metaDataSetName = metaDataSetName
+        self.dataSetDict = self._import_MERFISH_datasets(metaDataSetInfo)
+        self.metaDataSetParameters = self._import_parameters(metaDataSetInfo)
+        if 'terminal_tasks' in self.metaDataSetParameters:
+            for task in self.metaDataSetParameters['terminal_tasks']:
+                self._cache_aggregated_data(task)
+
+    def _import_MERFISH_datasets(self, MERFISHDataSets):
+        if not os.path.isfile(MERFISHDataSets):
+            sourcePath = os.sep.join([merlin.METADATA_HOME, MERFISHDataSets])
+        else:
+            sourcePath = MERFISHDataSets
+        destPath = os.sep.join([self.analysisPath, 'metadatasets.json'])
+        shutil.copyfile(sourcePath, destPath)
+
+        allDataSets = dict()
+        with open(destPath, 'r') as f:
+            mDataSets = json.load(f)['datasets']
+            for ds in mDataSets:
+                allDataSets[ds['dataset']] = ds['type']
+        return allDataSets
+
+    def _import_parameters(self, MERFISHDataSets):
+        path = os.sep.join([merlin.METADATA_HOME, MERFISHDataSets])
+
+        with open(path, 'r') as f:
+            return json.load(f)['metaparameters']
+
+    def _get_dataset_tasks(self, ds):
+        outDict = dict()
+        dirList = os.listdir(ds.analysisPath)
+        dirList = [x for x in dirList if
+                   os.path.isdir(os.sep.join([ds.analysisPath, x]))]
+        dirList = [x for x in dirList if x[0].isupper()]
+        for dir in dirList:
+            taskFile = os.sep.join([ds.analysisPath, dir, 'tasks', 'task.json'])
+            if os.path.isfile(taskFile):
+                with open(taskFile, 'r') as taskOpen:
+                    loadedTask = json.load(taskOpen)
+                    module = loadedTask['module']
+                    cl = loadedTask['class']
+                if module in outDict:
+                    outDict[cl] = [outDict[cl]] + [module]
+                else:
+                    outDict[cl] = [module]
+        return outDict
+
+    def _load_aggregated_data(self, analysisName: str):
+        allAnalyses = []
+        for k, v in self.dataSetDict.items():
+            ds = MERFISHDataSet(k)
+            requestedAnalyses = self._get_dataset_tasks(ds)
+            if analysisName in requestedAnalyses:
+                try:
+                    analysis = ds.load_analysis_task(analysisName)
+                    analysisResult = analysis.return_exported_data()
```

Comment:
I prefer to leave this aggregation and caching to an analysis task for two reasions. First, creating another place where the aggregated is stored could cause confusion about which version of the data is actually be analyzed if tasks within the MERFISHDataSet are rerun. Second, this is too specialized towards clustering. There are other meta analysis that can be be added in the future that are distinct from clustering.

My impression is that you wrote it this way to allow you to move the completed analysis tasks to a new location that is no longer accessible so that you can free up space. My solution to this problem is to create a skeleton of the analysis results where the tasks that take the largest space (such as Decode) are removed. This would allow all analysis results to be kept in easily accessible storage indefinitely.

Reply:
It was partly based on what you said, and partly because it became complicated to allow tasks to have dependencies that were outside their own dataset structure. For the aggregation task, for example, the dependencies are actually within the datasets that comprise the metadataset. I've made a handful of changes to enable this, primarily by enabling you to pass either a list of tasks or a list of task names as dependencies.